packages/peregrine/lib/talons/RootComponents/Product/useProduct.js

```diff
@@ -29,14 +34,23 @@ export const useProduct = props => {
     });
 
     const product = useMemo(() => {
+        if (!data) {
+            // The product isn't in the cache and we don't have a response from GraphQL yet.
+            return null;
+        }
+
+        const supportedProducts = data.products.items.filter(
+            isSupportedProductType
+        );
         // If a product is out of stock _and_ the backend specifies not to
         // display OOS items, the items array will be empty.
-        if (data && data.products.items[0]) {
-            return mapProduct(data.products.items[0]);
+        // Additionally, we may have an empty array after filtering out unsupported product types.
+        if (!supportedProducts.length) {
+            return null;
         }
 
-        // The product isn't in the cache and we don't have a response from GraphQL yet.
-        return null;
+        // Pick the first supported product.
+        return mapProduct(supportedProducts[0]);
```

Comment:
I can't believe `mapProduct` is still around, it was a compatibility fix for `2.3.1` which is long out of support. I'll create a tech debt task to remove it, which should clean this talon up a little bit.